### PR TITLE
FIX: executemany SQL_C_NUMERIC mismatch

### PR DESCRIPTION
**Pull request by jahnvi480**
### Work Item / Issue Reference  

> [AB#45380](https://sqlclientdrivers.visualstudio.com/c6d89619-62de-46a0-8b46-70b92a84d85e/_workitems/edit/45380)

> GitHub Issue: #609 

-------------------------------------------------------------------
### Summary   
This pull request addresses a critical bug in the handling of `executemany` with large `Decimal` values in the `mssql_python` driver, specifically when values exceed the SQL Server `MONEY` range. The main fix ensures that parameter type detection and conversion are consistent, preventing runtime errors when binding large decimal values. Extensive unit and integration tests are added to verify the fix and cover edge cases involving `Decimal` values, including scenarios with `NULL`s and multi-column inserts.

**Bug Fix: Executemany Decimal Handling**

* In `cursor.py`, the `executemany` method now explicitly overrides the C type for parameters with SQL type `DECIMAL` or `NUMERIC` to use `SQL_C_CHAR` (string binding) when the data is converted to strings. This prevents mismatches that previously caused runtime errors when inserting large decimal values. The column size is also adjusted to fit the longest string representation.

**Testing: Unit and Integration Tests for Decimal Handling**

* Added comprehensive unit tests in `test_001_globals.py` to verify type detection, mapping, and the override logic for `executemany` with large `Decimal` values, both within and outside the `MONEY` range. These tests confirm that the C type override is necessary and correctly applied.
* Added integration tests in `test_004_cursor.py` to exercise the fixed behavior in real database scenarios, including:
  - Inserting batches with decimals inside and outside the `MONEY` range.
  - Handling `NULL` values alongside large decimals.
  - Multi-column inserts where one column contains large decimals.

### CODE REVIEW

**Review by bewithgaurav — DISMISSED** [review]
reproduced #609 against sql2022, fix is clear and well-scoped. lgtm.

File: tests/test_001_globals.py [hunk 1]
```diff
@@ -1059,3 +1059,164 @@ def test_row_string_key_case_insensitive_with_lowercase():
...
+# GH-609: Unit tests for executemany SQL_C_NUMERIC mismatch
+# ---------------------------------------------------------
+import decimal as _decimal
+from mssql_python.cursor import Cursor, MONEY_MAX
+from mssql_python.constants import ConstantsDDBC as _C
+
+
+def _make_bare_cursor():
+    """Create a Cursor instance without a connection for unit testing."""
+    cur = Cursor.__new__(Cursor)
```
Comment (bewithgaurav) [inline tests/test_001_globals.py]: [non-blocking] suggestion to not have cursor tests in global test file

Comment (jahnvi480) [reply tests/test_001_globals.py]: Moved the tests to cursor

**Review by subrata-ms — APPROVED** [review]

**Review by bewithgaurav — APPROVED** [review]
